src/include/duckdb/common/limits.hpp

```diff
@@ -14,132 +14,237 @@ namespace duckdb {
+	DUCKDB_API static inline hugeint_t Maximum();
+	DUCKDB_API static inline bool IsSigned() {
 		return true;
 	}
-	DUCKDB_API static idx_t Digits() {
+	DUCKDB_API static inline idx_t Digits() {
 		return 39;
 	}
 };
 template <>
 struct NumericLimits<uint8_t> {
-	DUCKDB_API static uint8_t Minimum();
-	DUCKDB_API static uint8_t Maximum();
-	DUCKDB_API static bool IsSigned() {
+	DUCKDB_API static inline uint8_t Minimum();
+	DUCKDB_API static inline uint8_t Maximum();
+	DUCKDB_API static inline bool IsSigned() {
 		return false;
 	}
-	DUCKDB_API static idx_t Digits() {
+	DUCKDB_API static inline idx_t Digits() {
 		return 3;
 	}
 };
 template <>
 struct NumericLimits<uint16_t> {
-	DUCKDB_API static uint16_t Minimum();
-	DUCKDB_API static uint16_t Maximum();
-	DUCKDB_API static bool IsSigned() {
+	DUCKDB_API static inline uint16_t Minimum();
+	DUCKDB_API static inline uint16_t Maximum();
+	DUCKDB_API static inline bool IsSigned() {
 		return false;
 	}
-	DUCKDB_API static idx_t Digits() {
+	DUCKDB_API static inline idx_t Digits() {
 		return 5;
 	}
 };
 template <>
 struct NumericLimits<uint32_t> {
-	DUCKDB_API static uint32_t Minimum();
-	DUCKDB_API static uint32_t Maximum();
-	DUCKDB_API static bool IsSigned() {
+	DUCKDB_API static inline uint32_t Minimum();
+	DUCKDB_API static inline uint32_t Maximum();
+	DUCKDB_API static inline bool IsSigned() {
 		return false;
 	}
-	DUCKDB_API static idx_t Digits() {
+	DUCKDB_API static inline idx_t Digits() {
 		return 10;
 	}
 };
 template <>
 struct NumericLimits<uint64_t> {
-	DUCKDB_API static uint64_t Minimum();
-	DUCKDB_API static uint64_t Maximum();
-	DUCKDB_API static bool IsSigned() {
+	DUCKDB_API static inline uint64_t Minimum();
+	DUCKDB_API static inline uint64_t Maximum();
+	DUCKDB_API static inline bool IsSigned() {
 		return false;
 	}
-	DUCKDB_API static idx_t Digits() {
+	DUCKDB_API static inline idx_t Digits() {
 		return 20;
 	}
 };
 template <>
 struct NumericLimits<float> {
-	DUCKDB_API static float Minimum();
-	DUCKDB_API static float Maximum();
-	DUCKDB_API static bool IsSigned() {
+	DUCKDB_API static inline float Minimum();
+	DUCKDB_API static inline float Maximum();
+	DUCKDB_API static inline bool IsSigned() {
 		return true;
 	}
-	DUCKDB_API static idx_t Digits() {
+	DUCKDB_API static inline idx_t Digits() {
 		return 127;
 	}
 };
 template <>
 struct NumericLimits<double> {
-	DUCKDB_API static double Minimum();
-	DUCKDB_API static double Maximum();
-	DUCKDB_API static bool IsSigned() {
+	DUCKDB_API static inline double Minimum();
+	DUCKDB_API static inline double Maximum();
+	DUCKDB_API static inline bool IsSigned() {
 		return true;
 	}
-	DUCKDB_API static idx_t Digits() {
+	DUCKDB_API static inline idx_t Digits() {
 		return 250;
 	}
 };
 
 } // namespace duckdb
+
+/// FIXME
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/windows_undefs.hpp"
+#include <limits>
+
+namespace duckdb {
+
+using std::numeric_limits;
+
+int8_t NumericLimits<int8_t>::Minimum() {
```

Comment:
Not sure why these were in a separate .cpp file, maybe because they have to be exported as `DUCKDB_API`? Normally I would expect them to be constexpr.

Reply:
It would indeed be better for them to be `constexpr` - if you can get this working and compiling on all systems then that would be a welcome addition. Considering that might take some time with regards to CI debugging, I would request you to do this using [your fork's CI](https://github.com/akuzm/duckdb/actions) rather than using the shared CI on the main repository.

Reply:
> I would request you to do this using [your fork's CI](https://github.com/akuzm/duckdb/actions) rather than using the shared CI on the main repository.

Oh, nice, didn't know I could do that. Will try.